java/parquet/src/main/java/sleeper/io/parquet/record/ParquetRecordWriterFactory.java

```diff
@@ -0,0 +1,116 @@
+    }
+
+    public static ParquetWriter<Record> createParquetRecordWriter(Path path, Schema schema) throws IOException {
+        return new Builder(path, SchemaConverter.getSchema(schema), schema).build();
+    }
+
+    public static ParquetWriter<Record> createParquetRecordWriter(Path path, TableProperties tableProperties, Configuration conf) throws IOException {
+        return createParquetRecordWriter(path,
+            tableProperties.getSchema(),
+            tableProperties.get(COMPRESSION_CODEC),
+            tableProperties.getLong(ROW_GROUP_SIZE),
+            tableProperties.getInt(PAGE_SIZE),
+            tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_ROW_KEY_FIELDS),
+            tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_SORT_KEY_FIELDS),
+            tableProperties.getBoolean(DICTIONARY_ENCODING_FOR_VALUE_FIELDS),
+            conf);
+    }
+
+    public static ParquetWriter<Record> createParquetRecordWriter(Path path, Schema schema, Configuration conf) throws IOException {
+        TableProperties tableProperties = new TableProperties(new InstanceProperties());
+        tableProperties.setSchema(schema);
+        return createParquetRecordWriter(path, tableProperties, conf);
+    }
+
+    public static ParquetWriter<Record> createParquetRecordWriter(Path path,
+            Schema schema,
+            String compressionCodec,
+            long rowGroupSize,
+            int pageSize,
+            boolean dictionaryEncodingForRowKeyFields,
+            boolean dictionaryEncodingForSortKeyFields,
+            boolean dictionaryEncodingForValueFields,
+            Configuration conf) throws IOException {
+        Builder builder = new Builder(path, SchemaConverter.getSchema(schema), schema)
+            .withCompressionCodec(CompressionCodecName.fromConf(compressionCodec.toUpperCase(Locale.ROOT)))
+            .withRowGroupSize(rowGroupSize)
+            .withPageSize(pageSize)
+            .withConf(conf);
+
+        setDictionaryEncoding(builder, schema.getRowKeyFieldNames(), dictionaryEncodingForRowKeyFields);
+        setDictionaryEncoding(builder, schema.getSortKeyFieldNames(), dictionaryEncodingForSortKeyFields);
+        setDictionaryEncoding(builder, schema.getValueFieldNames(), dictionaryEncodingForValueFields);
+
+        return builder.build();
+    }
+
+    private static void setDictionaryEncoding(Builder builder, List<String> fieldNames, boolean dictionaryEncodingEnabled) {
+        for (String fieldName : fieldNames) {
+            builder = builder.withDictionaryEncoding(fieldName, dictionaryEncodingEnabled);
+        }
+    }
+
+    public static class Builder extends ParquetWriter.Builder<Record, Builder> {
+        private final MessageType messageType;
+        private final Schema schema;
+
+        public Builder(Path path, MessageType messageType, Schema schema) {
+            super(path);
+            this.messageType = messageType;
+            this.schema = schema;
+        }
```

Comment:
This constructor could be private, or it could just take the path and schema and infer the message type.